### PR TITLE
Allowing more flexible precision for TIMESTAMP, DATETIME fields.

### LINKED ISSUE #22558: [Bug]: BQ Storage Write API DateTime space formatter only allows for microsecond precision
### What happened?

Writing to BigQuery using Storage Write API with a TIMESTAMP field produces an error if values don't have microsecond precision. An example of a valid value that would throw an error: `2022-07-28 03:16:24.944`.

### Issue Priority

Priority: 3

### Issue Component

Component: io-java-gcp

### PR DESCRIPTION
**Pull request by ahmedabu98**
Fixes #22558

### CODE REVIEW

**Comment by ahmedabu98** [comment]
Run PostCommit_Java_DataflowV2

**Comment by ahmedabu98** [comment]
Run PostCommit_Java_DataflowV2

**Comment by ahmedabu98** [comment]
R: @reuvenlax

**Comment by ahmedabu98** [comment]
R: @chamikaramj

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java [hunk 1]
```diff
@@ -73,7 +74,12 @@ public class TableRowToStorageApiProto {
   // The old dremel parser accepts this format, and so does insertall. We need to accept it
   // for backwards compatibility, and it is based on UTC time.
   private static final DateTimeFormatter DATETIME_SPACE_FORMATTER =
-      DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSSSS").withZone(ZoneOffset.UTC);
+      new DateTimeFormatterBuilder()
```
Comment (johnjcasey) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java:77]: Is there a named datetime format for this we could use instead?

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProtoTest.java [hunk 2]
```diff
@@ -534,6 +556,7 @@ public void testNestedFromTableSchema() {
                   new TableCell().setV("1970-01-01T00:00:00.000+01:00"),
                   new TableCell().setV("1234567"),
                   new TableCell().setV("1970-01-01 00:00:00.000343"),
+                  new TableCell().setV("1970-01-01 00:00:00.123"),
```
Comment (johnjcasey) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProtoTest.java:559]: can we also have a trailing 0 test case?

**Review by johnjcasey — CHANGES_REQUESTED** [review]

**Comment by reuvenlax** [comment]
R: @yirutang could you take a look?

**Review by yirutang — APPROVED** [review]

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProtoTest.java (hunk 2 above)
Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProtoTest.java:575]: Done

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java (hunk 1 above)
Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java:77]: No readily available format for the `' '` character. The closest thing is [ISO_LOCAL_DATE_TIME](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_LOCAL_DATE_TIME) but that's for `'T'` between the date and time.

**Pull request merged**